parsec/mca/device/device_gpu.c

```diff
@@ -1700,7 +1717,7 @@ parsec_device_callback_complete_push(parsec_device_gpu_module_t   *gpu_device,
                     /* Nobody is at the door to handle that event on the source of that data...
                      * we do the command directly */
                     parsec_atomic_lock( &source->original->lock );
-                    int readers = parsec_atomic_fetch_sub_int32(&source->readers, 1);
+                    int readers = parsec_atomic_fetch_sub_int32(&source->readers, 1) - 1;
```

Comment:
Curious: does that mean we were leaking items here because we never pushed them back to the LRU?

Reply:
In the (somewhat rare) case when there was no GPU-manager for the source GPU, yes, we would leak some memory allocated via the `zone_malloc()`. The checks at termination would then complain, which is how we found the leak, and why we added the `zone_debug()` call at the completion checks in debug mode.